packages/data/src/components/use-select/index.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * WordPress dependencies
+ */
+import { createQueue } from '@wordpress/priority-queue';
+import {
+	useLayoutEffect,
+	useRef,
+	useMemo,
+	useEffect,
+	useReducer,
+} from '@wordpress/element';
+import { isShallowEqualObjects } from '@wordpress/is-shallow-equal';
+
+/**
+ * Internal dependencies
+ */
+import useRegistry from '../registry-provider/use-registry';
+import useAsyncMode from '../async-mode-provider/use-async-mode';
+
+/**
+ * Favor useLayoutEffect to ensure the store subscription callback always has
+ * the selector from the latest render. If a store update happens between render
+ * and the effect, this could cause missed/stale updates or inconsistent state.
+ *
+ * Fallback to useEffect for server rendered components because currently React
+ * throws a warning when using useLayoutEffect in that environment.
+ */
+const useIsomorphicLayoutEffect =
```

Comment:
I re-implemented this.  I realize that GB as a whole isn't really supporting server rendering but with this being a separate package that _could_ be used in a server rendering environment, this might be a valid thing to do (and follows a pattern that Redux is using for their hooks)